### DISCUSSION TITLE
Mark read messages and purge them

### DISCUSSION THREAD

**Discussion by kharabela** [nats-io/nats-server#5654] [Q&A] [answered]
I am able to publish and consume messages.
Is there a way to mark read messages and purge them after reading?.
I want to read only unmarked messages from stream.

This is how I am currently reading & unmarshalling  messages

`account := AccountInfoRecord{}
	cctx, err := consumer.Consume(func(msg jetstream.Msg) {
		log.Printf("Received Message: %s", string(msg.Subject()))
		log.Printf("Received Message Data: %s", string(msg.Data()))
		msg.Ack()
		err = json.Unmarshal([]byte(msg.Data()), &account)
		if err != nil {
			log.Fatal(err)
		}
		log.Printf("Consumer Data =>  Page: %s  -  Name: %s\n", account.PageName, account.Element.FirstName)
	})
`

**Comment by jnmoyne** [comment, marked as answer]
Yes, you should look into the "working queue" (and "interest") retention policies.

See: https://docs.nats.io/nats-concepts/jetstream/streams#retentionpolicy
